Dockerfile

```
@@ -0,0 +1,54 @@
+FROM golang:${GO_VERSION}-alpine AS go-builder
+WORKDIR /build
+COPY backend/go.mod backend/go.sum ./
+RUN go mod download
+COPY backend/ ./
+RUN CGO_ENABLED=0 go build -ldflags='-s -w' -o /out/tracera-backend ./cmd/main.go 2>/dev/null || \
+    CGO_ENABLED=0 go build -ldflags='-s -w' -o /out/tracera-backend .
+
+FROM node:${NODE_VERSION}-alpine AS frontend-builder
+WORKDIR /build
+COPY frontend/package.json frontend/bun.lockb* ./
+RUN corepack enable && bun install --frozen-lockfile || npm ci
```

Hadolint (2.14.0)

[info] Note that A && B || C is not if-then-else. C may run when A is true.

(SC2015)